libs/common/NumUtils.cs

```diff
@@ -113,14 +113,20 @@ public static long BytesToLong(int length, byte* source)
         /// <returns>True if sequence contains only numeric digits, otherwise false</returns>
         public static bool TryBytesToLong(int length, byte* source, out long result)
         {
```

Comment:
This method needs to be replaced with Utf8Parser

```suggestion
                {
            var fNeg = *source == '-';
            var beg = fNeg ? source + 1 : source;
            var len = fNeg ? length - 1 : length;
            result = 0;

            // Do not allow leading zeros
            if (len > 1 && *beg == '0')
                return false;

            // Parse number and check consumed bytes to avoid alphanumeric strings
            if (!Utf8Parser.TryParse(new Span<byte>(beg, len), out result, out var bytesConsumed) || bytesConsumed != len)
                return false;

            // Negate if parsed value has a leading negative sign
            result = fNeg ? -result : result;
            return true;
        }
```

Reply:
If we can go Utf8Parser, we could just entirely use Utf8Parser for everything in NumUtils?

Reply:
There is no official guideline yet. We are currently working on making iterative changes to the NumUtils methods and evaluate their performance impact.